multicluster/controllers/multicluster/memberclusterannounce_controller.go

```diff
@@ -114,28 +110,26 @@ func (r *MemberClusterAnnounceReconciler) Reconcile(ctx context.Context, req ctr
 		klog.V(2).InfoS("Reset lastUpdateTime", "cluster", memberAnnounce.ClusterID)
 		// Reset lastUpdateTime and connectedLeader for this member.
 		data.lastUpdateTime = time.Now()
-		if len(memberAnnounce.LeaderClusterID) == 0 {
-			data.leaderStatus.connectedLeader = v1.ConditionUnknown
-			data.leaderStatus.message = "Not connected to leader yet"
-			data.leaderStatus.reason = ""
+		data.leaderStatus.connectedLeader = v1.ConditionFalse
+		data.leaderStatus.message = fmt.Sprintf("Local cluster is not the leader of member: %v",
+			memberAnnounce.ClusterID)
+		data.leaderStatus.reason = ReasonNotLeader
+		// Check whether this local cluster is the leader for this member.
+		clusterClaim := &multiclusterv1alpha2.ClusterClaim{}
+		if err := r.Get(context.TODO(), types.NamespacedName{Name: multiclusterv1alpha2.WellKnownClusterClaimID, Namespace: req.Namespace}, clusterClaim); err == nil {
+			if clusterClaim.Value == memberAnnounce.LeaderClusterID {
+				data.leaderStatus.connectedLeader = v1.ConditionTrue
+				data.leaderStatus.message = fmt.Sprintf("Local cluster is the leader of member: %v",
+					memberAnnounce.ClusterID)
+				data.leaderStatus.reason = ReasonConnectedLeader
+			}
 		} else {
-			data.leaderStatus.connectedLeader = v1.ConditionFalse
-			data.leaderStatus.message = fmt.Sprintf("Local cluster is not the leader of member: %v",
-				memberAnnounce.ClusterID)
-			data.leaderStatus.reason = ReasonNotLeader
-			// Check whether this local cluster is the leader for this member.
-			clusterClaim := &multiclusterv1alpha2.ClusterClaim{}
-			if err := r.Get(context.TODO(), types.NamespacedName{Name: multiclusterv1alpha2.WellKnownClusterClaimID, Namespace: req.Namespace}, clusterClaim); err == nil {
-				if clusterClaim.Value == memberAnnounce.LeaderClusterID {
-					data.leaderStatus.connectedLeader = v1.ConditionTrue
-					data.leaderStatus.message = fmt.Sprintf("Local cluster is the leader of member: %v",
-						memberAnnounce.ClusterID)
-					data.leaderStatus.reason = ReasonConnectedLeader
-				}
+			if !apierrors.IsNotFound(err) {
+				return ctrl.Result{}, err
```

Comment:
We need to unlock. I suggest to set err to nil if apierrors.IsNotFound(err), and break and check err and return after line 135.

Reply:
Fixed, defined a variable `err`.